### PR TITLE
Customer balance reports

### PR DESCRIPTION
**Pull request by sauloperez**
#### What? Why?

Closes https://github.com/openfoodfoundation/openfoodnetwork/issues/6698.

This only fixes the OC management report by making it show the balance of each order instead of the customer's balance and using the new implementation, under a feature toggle as usual.

#### What should we test?

See _Acceptance Criteria & Tests_ at #6698

#### Release notes
Use new order balance calculation in order cycle management report but hidden under a feature toggle.

Changelog Category: User facing changes

### CODE REVIEW

**Review by andrewpbrett — APPROVED** [review]
This looks good! Straightforward and with a couple nice refactorings. 👍

File: spec/lib/open_food_network/order_cycle_management_report_spec.rb [hunk 1]
```diff
@@ -26,6 +26,14 @@ module OpenFoodNetwork
           expect(subject.orders).to eq([o2])
         end
 
+        it 'orders them by id' do
+          result = instance_double(ActiveRecord::Relation)
+          allow_any_instance_of(Ransack::Search).to receive(:result).and_return(result)
+          expect(result).to receive(:order).with(:id) { Spree::Order.none }
```
Comment (mkllnk) [inline spec/lib/open_food_network/order_cycle_management_report_spec.rb]: I think that this spec is too specific. I would create three orders with their ids in irregular order and then test that they are returned in the right order.

This spec would incorrectly pass if the code called `result.order(:completed_at).order(:id)`.

**Review by mkllnk — APPROVED** [review]
Nice work!

File: spec/lib/open_food_network/order_cycle_management_report_spec.rb (hunk 1 above)
Comment (sauloperez) [reply spec/lib/open_food_network/order_cycle_management_report_spec.rb]: Yes, it's very brittle so I might as well remove it. We can't fake ids to be in irregular order AFAIK. It's the db who assigns them and they'll always follow the order in which they were created.

Comment (sauloperez) [reply spec/lib/open_food_network/order_cycle_management_report_spec.rb]: We could as well orde by completed_at which might be better in terms of UX

Comment (mkllnk) [reply spec/lib/open_food_network/order_cycle_management_report_spec.rb]: > We can't fake ids to be in irregular order AFAIK.

We can! As long as there's no conflict, of course:

```
a = AdjustmentMetadata.order(:id).last
=> #<AdjustmentMetadata id: 1737, adjustment_id: 1739, enterprise_id: 3, fee_name: "markup", fee_type: "sales", enterprise_role: "coordinator">

b = AdjustmentMetadata.create(a.attributes.merge(id: a.id + 5))
=> #<AdjustmentMetadata id: 1742, adjustment_id: 1739, enterprise_id: 3, fee_name: "markup", fee_type: "sales", enterprise_role: "coordinator">
```

But I agree that ordering by another column is a much better idea. `created_at` is closest to `id` but maybe Rachel knows which default order the most sense.

Comment (sauloperez) [reply spec/lib/open_food_network/order_cycle_management_report_spec.rb]: Solved in the last commit.

**Review by mkllnk — APPROVED** [review]
Great! :tada:

**Comment by filipefurtad0** [comment]
Hey @sauloperez ,

This works like a charm. Below I'm comparing the customer balance reports, before and after toggling this for both the admin and the customer - I wanted to have a look at customer balance as well, while doing this. This list of orders includes cancelled and resumed orders. 

So, before toggling this PR:

image


After toggling this PR:

image

Before toggling, "cancelled and resumed" orders are displayed on this report, but not shown for the customer - the customer balances are correct though. After toggling, these orders are displayed for the customer as well.

Awesome, all good here :+1
Ready to go.

**Pull request merged**